ui/src/app/machines/views/MachineList/MachineListHeader/ActionFormWrapper/DeployForm/DeployFormFields/UserDataField/UserDataField.test.tsx

```diff
@@ -0,0 +1,244 @@
+import { act } from "react-dom/test-utils";
+import configureStore from "redux-mock-store";
+import { mount, ReactWrapper } from "enzyme";
+import { MemoryRouter } from "react-router-dom";
+import React from "react";
+import { Provider } from "react-redux";
+
+import DeployForm from "../../DeployForm";
+import { TSFixMe } from "app/base/types";
+
+const mockStore = configureStore();
+
+class MockFileReader {
+  result: string;
+  constructor() {
+    this.result = "test file content";
+  }
+  /* eslint-disable @typescript-eslint/no-empty-function */
+  onabort() {}
```

Comment:
Can you give these the type of `() => undefined` instead? TypeScript is trying to warn us here that we may have unintentionally left the function body undefined.

Reply:
Hmm this seems to be about leaving comments (https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-empty-function.md). I've added some comments and removed the eslint-disable.

Reply:
Does that seem right to you? I couldn't see a way to fix the error any other way...

Reply:
I think what ts is expecting is:

```typescript
  onabort = () => undefined;
  onerror = () => undefined;
  onload = () => undefined;
```

Reply:
Done.